### PR TITLE
Enable parallel runs of 64-bit XUnit

### PR DESCRIPTION
**Pull request by agocke**
This change enables us to run tests in parallel using new xunit2 64-bit runners.

The runner has to be 64-bit because otherwise we OOM. The new runtime with parallel enabled is ~12 minutes.

Some of the tests didn't support 64-bit running or parallel running for one reason or another. Where the fix was simple I did it myself. Otherwise, I filed a bug and disabled the test/assembly.

### CODE REVIEW

File: BuildAndTest.proj [hunk 1]
```diff
@@ -75,10 +69,9 @@
       <TestAssemblies Include="Binaries\Debug\Roslyn.Services.VisualBasic.UnitTests.dll" />
     </ItemGroup>
 
-    <xunit
-      Assemblies="@(TestAssemblies)"
-      Html="UnitTestResults.html"
-      ShadowCopy="false" />
+    <exec
+      Command="packages\xunit.runners.2.0.0-rc1-build2826\tools\xunit.console.exe @(TestAssemblies, ' ') -noshadow -parallel assemblies" />
+
```
Comment (jaredpar) [inline BuildAndTest.proj]: Do we have an issue to remove the assemblies argument to parallel?  Is it worth doing?

Comment (agocke) [inline BuildAndTest.proj]: I think we would need an issue just to investigate that, but I suspect we want to parallelize more. My CPU spends a fair amount of time idle when running the tests.

**Comment by jaredpar** [comment]
This is currently a PR into your own fork.  Did you mean to do a PR into dotnet/roslyn instead?

**Comment by agocke** [comment]
Unfortunately, this repo is a fork of your old git repo, so it doesn't recognize it as a fork of dotnet/roslyn. I just realized this myself :) I will delete this repository and restart on a fork.

**Pull request closed without merging**